src/components/SignupForm.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from 'react';
+import CryptoJS from 'crypto-js';
+import { useNavigate } from 'react-router-dom';
+import Email from './Email';
+import Password from './Password';
+import Button from './Button';
+import axios from 'axios';
+
+
+const SignupForm = () => {
+    const navigate = useNavigate();
+
+    const [email, setEmail] = useState('');
+    const [masterPassword, setMasterPassword] = useState('');
+    const [error, setError] = useState({});
+
+
+    const isValidEmail = (email) => {
+        const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+        return emailRegex.test(email);
+    };
+
+    const passwordPattern = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/;
+
+    const generateSalt = () => {
+        return CryptoJS.lib.WordArray.random(128 / 8).toString();
+    };
+
+    const hashPassword = (password, salt) => {
+        return CryptoJS.PBKDF2(password, salt, {
+            keySize: 256 / 32,
+            iterations: 1000,
+        }).toString();
+    };
+
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+        const newErrors = {};
+        setError(newErrors);
+
+        if (!isValidEmail(email)) {
+            newErrors.email = 'Please enter a valid email address.';
+        }
+
+        if (!masterPassword) {
+            newErrors.password = 'Password is required.';
+        } else if (!passwordPattern.test(masterPassword)) {
+            newErrors.password = 'Password must contain at least one uppercase letter, one lowercase letter, one digit, and one special character.';
+        }
+
+        if (Object.keys(newErrors).length > 0) {
+            setError(newErrors);
+            return;
+        }
+        const salt = generateSalt();
+        const hashedPassword = hashPassword(masterPassword, salt);
+
+        console.log("sending");
+        try {
+            const response = await axios.post('http://localhost:8080/api/register', {
```

Comment:
Here, the url `http://localhost:8080` should come from an env variable (you can check best practices to build a URL online).